amgut/lib/data_access/test/test_survey.py

```diff
@@ -0,0 +1,452 @@
+            'Pet_Information_127_0': 'MOV',
+            'Pet_Information_128_0': 1,
+            'Pet_Information_129_0': 2,
+            'Pet_Information_130_0': '6',
+            'Pet_Information_131_0': 1,
+            'Pet_Information_132_0': 1,
+            'Pet_Information_133_0': 3,
+            'Pet_Information_134_0': 2,
+            'Pet_Information_135_0': [1],
+            'Pet_Information_136_0': 3,
+            'Pet_Information_137_0': [0],
+            'Pet_Information_138_0': 1,
+            'Pet_Information_139_0': 3,
+            'Pet_Information_140_0': 3,
+            'Pet_Information_141_0': 4,
+            'Pet_Information_142_0': 'MOV',
+            'Pet_Information_143_0': 'MOV',
+            'Pet_Information_144_0': 'MOV',
+            'Pet_Information_145_0': 'Female: 50; Male: 59'})
+
+    def test_fetch_survey_bad_id(self):
+        survey = Survey(1)
+        with self.assertRaises(ValueError):
+            survey.fetch_survey('BAD_ID_HERE')
+
+    def test_store_survey(self):
+        # Create new survey ID and make sure it is unused
+        survey = Survey(2)
+        c = ascii_letters + '1234567890'
+        survey_id = ''.join([choice(c) for i in range(16)])
+        name = ''.join([choice(ascii_letters) for i in range(20)])
+        with self.assertRaises(ValueError):
+            survey.fetch_survey(survey_id)
+        consent = {
+            'login_id': 'd8592c74-84b1-2135-e040-8a80115d6401',
+            'survey_id': survey_id,
+            'participant_name': name,
+            'age_range': 'ANIMAL_SURVEY',
+            'parent_1_name': 'ANIMAL_SURVEY',
+            'parent_2_name': 'ANIMAL_SURVEY',
+            'deceased_parent': False,
+            'participant_email': 'REMOVED',
+            'obtainer_name': 'ANIMAL_SURVEY',
+            'assent_obtainer': 'ANIMAL_SURVEY',
+            'is_juvenile': True}
+        with_fk = [(survey_id, 128, 'Other'),
+                   (survey_id, 129, 'Wild'),
+                   (survey_id, 131, 'Male'),
+                   (survey_id, 132, 'Suburban'),
+                   (survey_id, 133, 'Normal'),
+                   (survey_id, 134, 'Omnivore'),
+                   (survey_id, 135, 'Wild food'),
+                   (survey_id, 136, 'Both'),
+                   (survey_id, 137, 'Unspecified'),
+                   (survey_id, 138, 'Lives alone with humans'),
+                   (survey_id, 139, '8+'),
+                   (survey_id, 140, 'Unspecified'),
+                   (survey_id, 141, 'Never')]
+        without_fk = [(survey_id, 130, '["20"]'),
+                      (survey_id, 142, '["Giant ratty pet!"]'),
+                      (survey_id, 143, '["Capybara"]'),
+                      (survey_id, 144, '[""]'),
+                      (survey_id, 145, '["29 - Male"]'),
+                      (survey_id, 127, '["%s"]' % name)]
+
+        survey.store_survey(consent, with_fk, without_fk)
+
+        obs = survey.fetch_survey(survey_id)
```

Comment:
The return types are from the database, so I'm not sure what you mean? The lists are from multiple answer questions or free text, and the rest are from controlled vocab that is already typed in the database.